cdap-ui/app/cdap/components/Reports/store/ReportsStore.js

```diff
@@ -196,7 +201,14 @@ const details = (state = defaultDetailsState, action = defaultAction) => {
     case ReportsActions.setRuns:
       return {
         ...state,
-        runs: action.payload.runs
+        runs: action.payload.runs,
+        totalRunsCount: action.payload.totalRunsCount,
+        totalRunsPages: Math.ceil(action.payload.totalRunsCount / state.runsLimit)
```

Comment:
This is not the case here, but what if multiple components need to use this value from the store? Would be cleaner to calculate in the store instead of doing it once for each component right?

Reply:
which other component will need this value?